packages/utilities/src/shadowDom/MergeStylesShadowRootContext.tsx

```diff
@@ -126,21 +127,40 @@ const adoptSheet = (
   shadowCtx: MergeStylesShadowRootContextValue,
   doc: Document,
   stylesheetKey: string,
-  stylesheet: CSSStyleSheet,
+  stylesheet: ExtendedCSSStyleSheet,
   listenerRef: PolyfileInsertListeners,
 ) => {
   const shadowRoot = shadowCtx.shadowRoot!;
 
   shadowCtx.stylesheets.set(stylesheetKey, stylesheet);
   const sheet = Stylesheet.getInstance();
   if (sheet.supportsConstructableStylesheets()) {
-    // The current spec allows the `adoptedStyleSheets` array to be modified.
-    // Previous versions of the spec required a new array to be created.
-    // For more details see: https://github.com/microsoft/fast/pull/6703
+    // Maintain the sort order of Fluent style sheets
+    const prevSheets = shadowRoot.adoptedStyleSheets;
+    let i = prevSheets.length;
+    let found = i === 0;
+    while (i >= 0 && !found) {
+      i--;
+
+      const prevSheet = prevSheets[i] as ExtendedCSSStyleSheet;
+      const prevSortOrder = (prevSheet.metadata?.sortOrder as number) ?? 0;
+      const sheetSortOrder = (stylesheet.metadata?.sortOrder as number) ?? 0;
+      if (prevSheet.bucketName === 'merge-styles' && prevSortOrder < sheetSortOrder) {
+        found = true;
+      }
+    }
+
     if (sheet.supportsModifyingAdoptedStyleSheets()) {
-      shadowRoot.adoptedStyleSheets.push(stylesheet);
+      // The current spec allows the `adoptedStyleSheets` array to be modified.
+      // Previous versions of the spec required a new array to be created.
+      // For more details see: https://github.com/microsoft/fast/pull/6703
+      shadowRoot.adoptedStyleSheets.splice(i + 1, 0, stylesheet);
```

Comment:
So, if I'm understanding correctly, this means that if the sheet was not found, it is essentially inserted at the front of the `adoptedStyleSheets` array, right?

Reply:
Yes. This is consistent with the existing `mergeStyles` behavior that will insert a sheet as the first child of `document.head` in this case.